### PR TITLE
packaging,tests: sync patches from debian

### PR DESCRIPTION
**Pull request by zyga**
This accounts for most of the changes we have in Debian. The remaining few patches are really Debian specific or probably paper over some bugs that we should not merge here.

### CODE REVIEW

**Review by sergiocazzolato — APPROVED** [review]
Thanks

File: packaging/debian-sid/snapd.lintian-overrides [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+# and precision as to what the actions mean is very important.
+snapd: maintainer-script-calls-systemctl [postrm:9]
+snapd: maintainer-script-calls-systemctl [prerm:9]
+# This could be possibly removed or migrated to a specific mechanism.
+snapd: maintscript-calls-ldconfig [postinst]
+# FIXME: Yes this one is missing but it is a programmatic interface that lives
+# in this location for complex reasons. We should write one and remove this
+# override.
+snapd: no-manual-page [usr/bin/snapctl]
+# TODO: ubuntu-core-launcher can be removed and is entirely obsolete.
```
Comment (zyga) [inline packaging/debian-sid/snapd.lintian-overrides:18]: I think this is gone now.

File: tests/lib/prepare-restore.sh [hunk 2]
```diff
@@ -269,6 +269,8 @@ prepare_project() {
     # no need to modify anything further for autopkgtest
     # we want to run as pristine as possible
     if [ "$SPREAD_BACKEND" = autopkgtest ]; then
+        create_test_user
+        systemctl enable --now snapd.socket
```
Comment (zyga) [inline tests/lib/prepare-restore.sh:273]: Debian CI still reports occasional failures. I don't quite know why but I suspect this is only partially sufficient.

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**